subprojects/soak/src/integTest/groovy/org/gradle/resolve/DependencyResolutionStressTest.groovy

```diff
@@ -99,74 +99,75 @@ task check {
+                        handleGetJar(response)
                         request.handled = true
-                    } else if (request.method == 'HEAD' && request.pathInfo == '/org.gradle/changing/1.0/changing-1.0.jar') {
-                        handleHeadJar(request, response)
+                    } else if (request.method == HEAD_METHOD && request.pathInfo == JAR_FILE_PATH) {
+                        handleHeadJar(response)
                         request.handled = true
                     }
                 }
             })
             server.start()
         }
 
-        private void handleGetIvy(HttpServletRequest request, HttpServletResponse response) {
+        @Override
+        protected void after() {
+            server.stop()
+        }
+
+        private void handleGetIvy(HttpServletResponse response) {
             println "* GET IVY FILE"
             def ivy = resources.ivy
-            response.setDateHeader(HttpHeaders.LAST_MODIFIED, ivy.lastModified)
-            response.setContentLength(ivy.contentLength)
-            response.setContentType(ivy.contentType)
+            setResourceInfo(response, ivy)
             ivy.writeContentTo(response.outputStream)
         }
 
-        private void handleHeadIvy(HttpServletRequest request, HttpServletResponse response) {
+        private void handleHeadIvy(HttpServletResponse response) {
             println "* HEAD IVY FILE"
-            def ivy = resources.ivy
-            response.setDateHeader(HttpHeaders.LAST_MODIFIED, ivy.lastModified)
-            response.setContentLength(ivy.contentLength)
-            response.setContentType(ivy.contentType)
+            setResourceInfo(response, resources.ivy)
         }
 
-        private void handleGetJar(HttpServletRequest request, HttpServletResponse response) {
+        private void handleGetJar(HttpServletResponse response) {
             println "* GET JAR"
             def jar = resources.jar
-            response.setDateHeader(HttpHeaders.LAST_MODIFIED, jar.lastModified)
-            response.setContentLength(jar.contentLength)
-            response.setContentType(jar.contentType)
+            setResourceInfo(response, jar)
             jar.writeContentTo(response.outputStream)
         }
 
-        private void handleHeadJar(HttpServletRequest request, HttpServletResponse response) {
+        private void handleHeadJar(HttpServletResponse response) {
             println "* HEAD JAR"
-            def jar = resources.jar
-            response.setDateHeader(HttpHeaders.LAST_MODIFIED, jar.lastModified)
-            response.setContentLength(jar.contentLength)
-            response.setContentType(jar.contentType)
+            setResourceInfo(response, resources.jar)
         }
 
-        @Override
-        protected void after() {
-            server.stop()
+        private void setResourceInfo(HttpServletResponse response, Resource resource) {
```

Comment:
I would call this something like `setReponseHeaders` or `provideHeadersForResource`.